### PR TITLE
Extract public

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Extract jflyte utility classes into a separated module.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description

 This is to make it easier to extend jflyte or replacing certain subcommands, otherwise things have to be done in `org.flyte.jflyte` packages.

Note that not everything in jflyte-utils is public because we only want to open up things we know for sure are already being used and we can open up others when needed.

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: jflyte-utils/pom.xml [hunk 1]
```diff
@@ -0,0 +1,182 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright 2021 Flyte Authors.
```
Comment (bergman) [inline jflyte-utils/pom.xml:3]:
```suggestion
  Copyright 2023 Flyte Authors.
```
Unless there is some specific requirement that it this license section is the exact same everywhere?

**Review by bergman — APPROVED** [review]
Had only a nitpick.

File: jflyte-utils/pom.xml (hunk 1 above)
Comment (bergman) [reply jflyte-utils/pom.xml:3]: Ah, I was afraid it was something like this. Thanks for confirming!
